mantidimaging/gui/windows/nexus_load_dialog/presenter.py

```diff
@@ -133,15 +133,17 @@ def _read_rotation_angles(self, image_key: int, before: bool | None = None) -> n
         """
         assert self.image_key_dataset is not None and self.rotation_angles is not None
         if before is None:
-            rotation_angles = self.rotation_angles[np.where(self.image_key_dataset[...] == image_key)]
+            image_key_array = np.array(self.image_key_dataset)
+            rotation_angles = np.array(self.rotation_angles)[np.where(image_key_array == image_key)]
```

Comment:
Can these be avoided by setting the types as `np.ndarray|None` in the constructor? That would allow the type check to catch anything assigning a different type to them.